### PR TITLE
Add overload to String.from_utf16 with pointer

### PR DESCRIPTION
**Pull request by straight-shoota**
This allows handling null-terminated wchar strings.

### CODE REVIEW

File: src/string/utf16.cr [hunk 1]
```diff
@@ -53,7 +53,15 @@ class String
...
   # String.from_utf16(slice) # => "hi 𐂥"
   # ```
-  def self.from_utf16(slice : Slice(UInt16)) : String
+  #
+  # If *slice* is a pointer, the string is interpreted as null-terminated.
+  #
+  # ```
+  # slice = Slice[104_u16, 105_u16, 32_u16, 0_u16, 55296_u16, 56485_u16]
+  # String.from_utf16(slice)           # => "hi \0000𐂥"
+  # String.from_utf16(slice.to_unsafe) # => "hi"
```
Comment (asterite) [inline src/string/utf16.cr]: I think the result is missing a space

File: src/string/utf16.cr [hunk 2]
```diff
@@ -53,7 +53,15 @@ class String
   # slice = Slice[104_u16, 105_u16, 32_u16, 55296_u16, 56485_u16]
   # String.from_utf16(slice) # => "hi 𐂥"
   # ```
-  def self.from_utf16(slice : Slice(UInt16)) : String
+  #
+  # If *slice* is a pointer, the string is interpreted as null-terminated.
```
Comment (asterite) [inline src/string/utf16.cr]: The input pointer could be interpreted as a string, but maybe it's better to say that it ends when a value of zero is found?

**Comment by straight-shoota** [comment]
merge?

**Comment by RX14** [comment]
Has it been tested on actual windows code?

**Pull request merged**